### PR TITLE
GCP gVNIC support

### PR DESCRIPTION
**Pull request by jepio**
# Add support for GCP gVNIC

This enables the gve driver in the kernel, which handles the gVNIC device on GCP. We also update existing networkd configs for gcp to handle gve devices in the same way as virtio_net so that device naming is predictable (eth0 with aliases).

- [ ] https://github.com/flatcar-linux/init/pull/67
- [ ] https://github.com/flatcar-linux/mantle/pull/322

## How to use

Together with mantle https://github.com/flatcar-linux/mantle/pull/322

## Testing done

Tested here:
http://jenkins.infra.kinvolk.io:8080/job/os/job/manifest/5373/cldsv/

- [ ] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)

### CODE REVIEW

File: changelog/updates/2022-04-07-google-gve.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- Enable GVE kernel option, which adds support for Google Virtual NIC on GCP (TODO: link)
```
Comment (pothos) [inline changelog/updates/2022-04-07-google-gve.md:1]:
```suggestion
- Enabled GVE kernel driver, which adds support for Google Virtual NIC on GCP (TODO: link)
```
Maybe this file should be under `changes`, or?

Comment (jepio) [reply changelog/updates/2022-04-07-google-gve.md]: Reworded and moved to changes.

File: changelog/changes/2022-04-07-google-gve.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- Enabled GVE kernel driver, which adds support for Google Virtual NIC on GCP ([PR#1802](https://github.com/flatcar-linux/coreos-overlay/pull/1802))
```
Comment (pothos) [inline changelog/changes/2022-04-07-google-gve.md:1]:
```suggestion
- GCE: Enabled GVE kernel driver, which adds support for Google Virtual NIC on GCP ([PR#1802](https://github.com/flatcar-linux/coreos-overlay/pull/1802))
```
Release notes are skimmed and I think it helps to have the OEM prefixes

**Review by pothos — APPROVED** [review]

File: changelog/changes/2022-04-07-google-gve.md (hunk 2 above)
Comment (jepio) [reply changelog/changes/2022-04-07-google-gve.md]: will take note for the future.

**Pull request merged**